internal/querycoordv2/job/job.go

```diff
@@ -0,0 +1,487 @@
+package job
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/proto/querypb"
+	"github.com/milvus-io/milvus/internal/querycoordv2/meta"
+	"github.com/milvus-io/milvus/internal/querycoordv2/session"
+	"github.com/milvus-io/milvus/internal/querycoordv2/utils"
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+	"github.com/samber/lo"
+	"go.uber.org/zap"
+)
+
+// Job is request of loading/releasing collection/partitions,
+// the execution flow is:
+// 1. PreExecute()
+// 2. Execute(), skip this step if PreExecute() failed
+// 3. PostExecute()
+type Job interface {
+	MsgID() int64
+	CollectionID() int64
+	// PreExecute does checks, DO NOT persists any thing within this stage,
+	PreExecute() error
+	// Execute processes the request
+	Execute() error
+	// PostExecute clears resources, it will be always processed
+	PostExecute()
```

Comment:
PreExecute, Execute, and PostExecute should belong to an executor Interface, maybe we cloud consider the interface design, make it clear.